parser/parser.go

```diff
@@ -37,6 +37,28 @@ func isDirectory(path string) (bool, error) {
 	return fileInfo.IsDir(), nil
 }
 
+func ParseSchemaFile(file string) (*ast.Schema, error) {
+	contents, err := os.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+	source := &ast.Source{Name: file, Input: string(contents)}
+	schema, schemaerr := gql.LoadSchema(source)
+	if schemaerr != nil {
+		return nil, schemaerr
+	}
+	return schema, nil
+}
+
+func ParseSchema(sources ...*ast.Source) (*ast.Schema, error) {
```

Comment:
The plan is that `ParseSchema` will be the only pub exposed method by `parser`. `ParseQuerySource` and `ParseSchemaFile` will be merged into `Schema` and used through it instead. However there's ongoing work related to this too so I'm not doing that as part of this PR.

Reply:
hm interesting, do we expect many sources?

Reply:
Yes! Depending on the source (file or download) we need to provide different sources. For download sources we need to _skip_ the prelude (built in default types) as they are included in the JSON response. However for files (SDL) these types are NOT included. In addition there's a "problem" with Absinthe servers that doesn't include `directive @deprecated` so we add that in as a source too when needed. Hence we need to use the `LoadSchema` call where we specify the sources ourselves.